modules/generator/processor/localblocks/livetraces.go

```diff
@@ -39,21 +36,16 @@ func (l *liveTraces) Len() uint64 {
 	return uint64(len(l.traces))
 }
 
-func (l *liveTraces) Push(batch *v1.ResourceSpans, max uint64) error {
-	if len(batch.ScopeSpans) == 0 || len(batch.ScopeSpans[0].Spans) == 0 {
-		return nil
-	}
-
-	traceID := batch.ScopeSpans[0].Spans[0].TraceId
+func (l *liveTraces) Push(traceID []byte, batches []*v1.ResourceSpans, maxTraces uint64) bool {
```

Comment:
Can we assert that the `batches` here contains only spans with the same traceID?

Reply:
Good catch. It actually doesn't because I missed that the distributor [regroups the batches across traces]
(https://github.com/grafana/tempo/blob/main/modules/distributor/distributor.go#L460).  I'll fix.